### PR TITLE
Update README, containers, CI for software.eessi.io

### PR DESCRIPTION
**Pull request by bedroge**
Replaced all paths / names of the old `pilot.eessi-hpc.org` repo to the new `software.eessi.io` repo, and removed "pilot" in any references to "pilot stack" (also in container image names). Also bumped some versions of tools used in the containers (CVMFS, AWSCLI).

~TODO: don't install the EESSI CVMFS configuration package, but instead check if the repo can be accessed using `cvmfs-config-default`; this cannot be done though, as the EESSI configuration is not part of it yet (probably as of Monday).~

The containers now include the new EESSI CVMFS configuration package that has the configuration for both the old and new domain.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -239,17 +240,17 @@ X-Cache: HIT from url-to-your-proxy
 
 Example with the Norwegian Stratum 1:
 ```
-curl --head http://bgo-no.stratum1.cvmfs.eessi-infra.org/cvmfs/pilot.eessi-hpc.org/.cvmfspublished
+curl --head http://bgo-no.stratum1.cvmfs.eessi-infra.org/cvmfs/software.eessi.io/.cvmfspublished
```
Comment (trz42) [inline README.md:243]: This doesn't work for `software.eessi.io`. Just a configuration issue of the stratum 1?

**Review by trz42 — CHANGES_REQUESTED** [review]
Looks good. One `curl` command to (test) access a file in the new EESSI stack (`software.eessi.io`) via the Norwegian Stratum-1 doesn't work. Likely a configuration issue on that Stratum-1. Maybe it could be replaces with another Stratum-1 (or temporarily removed/commented).

File: README.md (hunk 1 above)
Comment (bedroge) [reply README.md]: Well spotted! Fixed it by using a new Stratum 1.

**Review by trz42 — APPROVED** [review]
Works. :tada:

**Pull request merged**